### PR TITLE
Gracefully log local and closure object types

### PR DESCRIPTION
**Pull request by sprout42**
Some small changes to gracefully fail on local names and closures. the closures were the ones causing me the most trouble.

I'm not completely sure about how I implemented the cursor position save in `itanium_demangler/itanium_demangler/__init__.py`? that appears counter to how the tool seems to work. I tried to look into it further to figure out how to gracefully fail out without consuming the string when `_parse_special` found a situation it couldn't handle but in the time (and attention) I have I was not able to figure out a more graceful way to do that.

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -12,7 +12,9 @@
 
 import sys
 import os.path
-sys.path.append(os.path.join(os.path.dirname(__file__), "itanium_demangler"))
+# Prepend so if the itanium-demangler package is installed elsewhere it doesn't 
+# interfere
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), "itanium_demangler"))
```
Comment (whitequark) [inline __init__.py:17]: Please split this into a separate commit.

Comment (sprout42) [reply __init__.py:17]: Sorry for the delay, once I split it out and tested it more I realized that I wasn't handling things as graceful as I thought. it's a few commits now that hopefully are all distinct chunks of functionality/changes. But I have tested this on a few different C++ heavy binaries and it seems to work for the types that it supports and fails gracefully on demangler failures.

**Review by whitequark — APPROVED** [review]
LGTM; I have only a very limited context here but this all seems fine, and I assume you've tested it as well.

**Comment by whitequark** [comment]
Thanks, and apologies for the long delay!
